docs/proposals/koordlet/20220615-qos-manager.md

```diff
@@ -0,0 +1,97 @@
+---
+title: QoS Manager
+authors:
+  - "@stormgbs"
+reviewers:
+  - "@eahydra"
+  - "@hormes"
+  - "@jasonliu747"
+  - "@zwzhang0107"
+creation-date: 2022-06-15
+last-updated: 2022-06-15
+---
+
+# QoS Manager
+
+## Table of Contents
+
+<!--ts-->
+
+* [QoS Manager](#qos-manager)
+  * [Table of Contents](#table-of-contents)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals/Future Work](#non-goalsfuture-work)
+  * [Proposal](#proposal)
+    * [Design](#design)
+  * [References](#references)
+
+<!--te-->
+
+## Summary
+
+QoS Manager is a new plugin framework for `dynamic plugins` in [issue](https://github.com/koordinator-sh/koordinator/issues/174), `static plugins` will be move into [RuntimeHooks](https://github.com/koordinator-sh/koordinator/blob/main/docs/design-archive/koordlet-runtime-hooks.md).
+
+
+## Motivation
+
+Currently, plugins from resmanager in Koorlet are mixed together, they should be classified into two categories: `static` and `dynamic`. Static plugins will be called and run only once when container is created, updated, started or stopped. But for dynamic plugins, they may be called and run at any time according the real-time runtime state of node, such as CPU suppress, CPU burst, etc. This proposal only focuses on refactoring dynamic plugins. Take a look at current plugin implementation, there are many function calls to resmanager's methods directly, such as collecting node/pod/container metrics, fetching metadata of node/pod/container, fetching configurations(NodeSLO, etc.). In the feature, we may need a flexible and powerful famework with scalability for special external plugins.
+
+
+### Goals
+- Define a new framework named `qos-manager` to manage dynamic plugins.
+- Define new interfaces to abstract and simplify management of plugins.
+- Refactor and move the built-in dynamic plugins into `qos-manager`, such as `CPU suppress`, `CPU brust`, `CPU evict` and `memory evict`.
+
+### Non-Goals/Future Work
+
+- Framework design with scalability for external plugins.
+- Refctor `resource-update-executor`.
+
+## Proposal
+### Design
+
+![image](../../images/qos-manager.svg)
+
+ The below is directory tree of qos-manager inside koordlet, all existing dynamic pluins(as built-in plugins) will be move into sub-directory  `plugins`.
+ 
+```
+pkg/koordlet/qos-manager/
```

Comment:
we should constrain the style of package folder name, such as rename qos-mamager to qosmanager, cpu-burst to cpuburst